tests/test_utils.py

```diff
@@ -100,5 +101,24 @@ def test_compute_class_weights(self):
         self.assertEqual(weights[1], 1.0)
 
 
+class TestGradientAccumulation(unittest.TestCase):
+
+    def test_gradient_accumulating_optimizer(self):
+        loss = tf.get_variable("loss", shape=1)
+        opt = get_grad_accumulation_optimizer(tf.train.GradientDescentOptimizer, 2)(0.1)
+        train_op = opt.minimize(tf.abs(loss))
+        sess = tf.Session()
+        sess.run(tf.global_variables_initializer())
+        for i in range(100):
+            val_before = sess.run(loss)
+            sess.run(train_op)
+            val_after1 = sess.run(loss)
+            sess.run(train_op)
+            val_after2 = sess.run(loss)
+            self.assertEqual(val_before, val_after1)  # first step should not actually do anything
+            self.assertNotEqual(val_after1, val_after2)  # this should have applied a step and should be different.
+
+
```

Comment:
Another test I'd consider adding is to make sure the gradients are indeed being added (though see my main comment about adding vs. averaging). I believe it should be easy enough by replacing opt.minimize() with opt.compute_gradients() followed by opt.apply_gradients().